drivers/pci/pcie_brcmstb.c

```diff
@@ -628,6 +905,11 @@ static int brcm_pcie_of_to_plat(struct udevice *dev)
 	else
 		pcie->gen = max_link_speed;
 
+	pcie->pcie_cfg = (const struct brcm_pcie_cfg_data *)dev_get_driver_data(dev);
+
+	if (IS_ENABLED(CONFIG_BCM2712) && pcie->pcie_cfg->type == BCM2712)
```

Comment:
The idea was to leave CONFIG_BCM2712 option (as we have CONFIG_BCM2711 for rpi4, CONFIG_BCM2384 and so on) but make this option to indicate generic BCM2712 support. And it's purpose is to prevent unused parts to be linked to the code when BCM2712 is disabled( such ad resets in pcie or gpio in macb).
That's why this ifdef was used only in a few places. 
As for point 2 - Could you please elaborate a bit. I'm not sure I understand the purpose.

Reply:
The point is to have PCI_BRCMSTB_BCM2712 which belongs to "pci" and not to use some global config - tomorrow there could be BCM2713, but with the same PCIe.

And please or use ifdef everywhere or pcie->pcie_cfg->type == BCM2712 - you'r linker optimizations are partial which makes code looks very misleading. For example,` bcm2712_cfg` not under ifdef.